### PR TITLE
improve form buttons

### PR DESCRIPTION
**Pull request by jelhan**
* Fixes wrongly labeled submit button for participation (`next` to `save`). There might be still room for improvements (e.g. `participate` or `save participation`). Will leave that for another iteration.
* Adds a loading spinners to submit buttons while network request to create a poll or participate in a poll are pending.
* Some internal refactoring for better maintainability. Also introduces angle-bracket component invocation syntax. There is a very good page in Ember guides discussing the difference between these new syntax and the classical one: https://guides.emberjs.com/release/reference/syntax-conversion-guide/

I don't have a strong opinion on the concrete loading spinner being used. Just wanted to add one as not showing any reaction on user input is very bad UX.

@sappor0 would love to get your feedback as you have recently worked a lot on UI.

### CODE REVIEW

File: config/bundlesize.js [hunk 1]
```diff
@@ -4,12 +4,12 @@ module.exports = {
...
     javascript: {
       pattern: 'assets/*.js',
-      limit: '375KB',
+      limit: '376KB',
       compression: 'gzip'
     },
     css: {
       pattern: 'assets/*.css',
-      limit: '22KB',
+      limit: '15KB',
```
Comment (jelhan) [inline config/bundlesize.js]: #205 reduced size of CSS assets by nearly 1/3 (7 KB) after gzip. Done very well @sappor0! :clap:

**Comment by sappor0** [comment]
I think a loading spinner at the place where the interaction took place (the click on the button) is a great idea. I haven't managed to get it to show (throttling in my web dev tools didn't work for some reason, maybe because it's using a locally hosted backend) – how did you get it to show?

**Comment by sappor0** [comment]
Apart from that :+1: for the submit button label update, noticed that as well. IMHO making it more explicit isn't necessary because what is being saved is clear from what data the user just put in. (There are two scenarios, participation saving and poll creation – I think in both cases it should be clear what is being saved. – If we add poll editing I think it would make sense to add this distinction because it would be possible to mix them up)

**Comment by jelhan** [comment]
> I haven't managed to get it to show (throttling in my web dev tools didn't work for some reason, maybe because it's using a locally hosted backend) – how did you get it to show?

Did you run the local instance using `ember serve` without `proxy` option? In that case you are not using the API but a mock provided by a tool called [Mirage](https://www.ember-cli-mirage.com/). It intercepts all ajax and fetch requests using [Pretender](https://github.com/pretenderjs/pretender). Since the request is intercepted, there isn't any network request. Therefore it won't show up in network tab neither would be affected by throttle settings. But Mirage is configured to throttle all requests by 400ms: https://github.com/jelhan/croodle/blob/f10e6da6d65e5e9d8b227a14a2ff0f7f3cc082b4/mirage/config.js#L3 So it should show up for a very short time. You could adjust that config to make review easier.

If you are using `ember server --proxy http://127.0.0.1:8080` for local development Mirage is disabled. Requests should be shown in network tab and throttling by web dev tools should take affect. In that case I've no idea, what's going on.

**Comment by sappor0** [comment]
I tested it with the proxy option. I will try it with mirage. :)

**Comment by sappor0** [comment]
Got it to work.

Not sure if this is the correct way to do it but I opened a MR to this feature branch to use bootstrap loading spinners: https://github.com/jelhan/croodle/pull/216

**Comment by jelhan** [comment]
@sappor0 Thanks a lot for refactoring to use Bootstrap loading spinner. Is there anything else left that you would like to discuss / change before merging this one?

**Comment by sappor0** [comment]
No. Go for it :)

**Pull request merged**